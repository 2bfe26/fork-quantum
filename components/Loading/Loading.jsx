import React from 'react';

export default () => (
  <React.Fragment>
    Loading implementation
  </React.Fragment>
);
