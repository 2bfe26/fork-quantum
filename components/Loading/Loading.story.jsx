import React from 'react';
import { storiesOf } from '@storybook/react';
import HowToImport from '../../.storybook/decorators/HowToImport';
import Loading from './Loading';

storiesOf('6. Loading', module)
  .addDecorator(HowToImport)
  .add('Loading', () => (
    <React.Fragment>
      <Loading />
    </React.Fragment>
  ));
