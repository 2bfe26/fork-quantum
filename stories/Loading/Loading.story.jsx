import React from 'react';
import { storiesOf } from '@storybook/react';
import Heading from '../../.storybook/components/Heading';
import AutoPropsApi from '../../.storybook/components/AutoPropsApi';
import ComponentPanel from '../../.storybook/components/ComponentPanel';
import { TabbedView, Tab } from '../../.storybook/components/TabbedView';
import Loading from '../../components/Loading';
import Atom from '../static/atom.svg';

storiesOf('6. Others', module)
  .add('Loading', () => (
    <Heading image={Atom} title="<Loading />">
      <TabbedView>
        <Tab title="Usage">
          <ComponentPanel component={<Loading />} importModules="Loading" />
        </Tab>

        <Tab title="API">
          <AutoPropsApi component={Loading} />
        </Tab>
      </TabbedView>
    </Heading>
  ));
