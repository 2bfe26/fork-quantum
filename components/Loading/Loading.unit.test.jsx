import React from 'react';
import renderer from 'react-test-renderer';
import Loading from './Loading';

describe('Loading component ', () => {
  it('should match the snapshot', () => {
    expect(renderer.create(<Loading />).toJSON()).toMatchSnapshot();
  });

  it('should match the snapshot when it\'s not visible', () => {
    expect(renderer.create(<Loading visible={false} />).toJSON()).toMatchSnapshot();
  });
});
