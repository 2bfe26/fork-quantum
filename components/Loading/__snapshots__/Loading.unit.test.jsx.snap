// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component  should match the snapshot 1`] = `
<div
  className="Loading__Wrapper-sc-1jan2m7-0 kMhPws"
  size={60}
>
  <svg
    className="Loading__Svg-sc-1jan2m7-1 iZTpBD"
    viewBox="25 25 50 50"
  >
    <circle
      className="Loading__Circle-sc-1jan2m7-2 kERifh"
      cx="50"
      cy="50"
      r="12"
    />
  </svg>
</div>
`;

exports[`Loading component  should match the snapshot when it's not visible 1`] = `
<div
  className="Loading__Wrapper-sc-1jan2m7-0 jFGGzX"
  size={60}
>
  <svg
    className="Loading__Svg-sc-1jan2m7-1 iZTpBD"
    viewBox="25 25 50 50"
  >
    <circle
      className="Loading__Circle-sc-1jan2m7-2 kERifh"
      cx="50"
      cy="50"
      r="12"
    />
  </svg>
</div>
`;
